content/45.user_friendliness.md

```diff
@@ -1,7 +1,67 @@
+In this framing, library codes primarily interface with application codes, rather than with researchers.
+We have taken a slightly different approach with the design and development of yt, as it exists in a middle-ground as a library used *as an application* within the scripting language python and also as a library for more complex analysis of data.
+As an outgrowth of this, we have taken particular care with the "public-facing" API of yt.
+We have attempted to abstract the API enough from the data structures yt uses internally that it is useful without detailed knowledge of yt's internals.
+By the same token, we have also attempted to provide low-level access to data, and to make as many of those methods accessible and usable as well.
+
+In addition to API considerations, there are three areas that we note have additional care paid to the generation of figures, useful error messages, and to integration with the omnipresent Jupyter environment.
+
 ### Publication-Ready Figures
 
-### Jupyter Integration
+Matplotlib @doi:10.5281/zenodo.7570264 is a fully-featured mechanism for generating figures, with an incredible array of options to customize formatting, appearance, rendering of fonts and glyphs, sleection of colorbars, calculation of tick locations, and appropriate bounds.
+This degree of flexibility provides extremely fine-grained control over the appearance of figures for publication, and Matplotlib provides a large number of output formats for even the most discerning of journals.
+
+yt utilizes matplotlib as its primary rendering engine for visualizations; while 3D renderings (such as those described in @sec:vr) typically generate raw pixel buffers that are saved as images directly, most other visualization functionality in yt relies on matplotlib for generating images that are saved to disk.
+Slices and projections are prepared as variable-resolution images that are pixelized (@sec:pixelization) into fixed resolution buffers, then provided to matplotlib in the `imshow` command.
+Phase plots utilize the `pcolormesh` operation.
+
+All of these internal plots utilize matplotlib's "object-oriented" interface, wherein `Figure` and `Axes` objects are created directly.
+yt then manages the internal state of these objects, and provides high-level access to operations that are aware of the nature of the visualizations; for instance, operations that zoom and pan, and that are unit-aware.
+In the case of 2D spatial images, these are all generated as subclasses of a `PlotWindow` object, named such because it functions as a "window" onto the data, including automatically generating and managing multiple fields simultaneously.
+
+Within the `PlotWindow` object, we set up two levels of data invalidation; when a characteristic of the data is changed (i.e., the viewing window, the resolution) then the *data* is said to be invalidated.
```

Comment:
I've always found the term "data" to be pretty confusing in this context: it's not the simulation data itself that's invalidated, rather the _image_ data. I think this should be clarified